odoo/models.py

```diff
@@ -1746,6 +1747,108 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
+                                 Aug  Sep  Oct  Nov  Dec
+
+        :param list data: the data containing groups
+        :param list groupby: name of the first group by
+        :param aggregated_fields list: list of aggregated fields in the query
+        :param relativedelta interval: interval between to temporal groups
+        expressed as a relativedelta month by default
+        :rtype: list
+        :return: list
+        """
+        groupby_name = groupby[0]
+
+        # existing non null datetimes
+        existing = [d[groupby_name] for d in data if d[groupby_name]]
+
+        if len(existing) < 2:
+            return data
+
+        # assumption: existing data is sorted by field 'groupby_name'
+        first, last = existing[0], existing[-1]
+
+        # When the date_trunc function is used on a DATE (not a TIMESTAMP),
+        # PostGreSQL implicitly cast it as a TIMESTAMP WITH TIMEZONE. That time-
+        # zone is based on the local machine executing Postgres. We don't want
+        # this because it could cause various problems. That's why we detect
+        # those TZs by assuming they are always an instance of
+        # psycopg2.tz.FixedOffsetTimezone and remove them. Other possible
+        # timezone should come from the user context.
+        remove_tz = isinstance(first.tzinfo, psycopg2.tz.FixedOffsetTimezone)
+        if remove_tz:
+            first = first.replace(tzinfo=None)
+            last = last.replace(tzinfo=None)
+
+        expected = list(date_utils.date_range(first, last, interval))
+
+        if len(existing) < len(expected):
+            tz = data[0][groupby_name].tzinfo
+            empty_data = dict.fromkeys(aggregated_fields, False)
+            empty_data['id'] = False
+            empty_data[groupby_name.split(':')[0] + '_count'] = 0
+
+            if not data[-1][groupby_name]:
+                nulls = [data[-1]]
+                data = collections.deque(data[:-1])
+            else:
+                nulls = []
+                data = collections.deque(data)
+
+            new_data = []
+
+            # Note: the list 'expected' contains the the dates that should be
+            # represented inside data when it is returned. Notice that 'expected'
+            # is sorted like data, so what we do is empty data progressivly by
+            # popping its elements from first to last. To do this properly, we
+            # need to compare dates together, but here comes a little problem :
+            # in some circumstances, dates can have a different timezone resulting
+            # in a false negative. To avoid that, we remove that timezone but it
+            # could be an error.
+            for dt in expected:
+                if remove_tz:
+                    data[0][groupby_name] = data[0][groupby_name].replace(tzinfo=None)
+
+                if data[0][groupby_name] == dt:
+                    new_data.append(data.popleft())
+                else:
```

Comment:
we can simplify and preserve NULLS ordering by modifying as follows, without needing `nulls` etc:

```
if not data[0][groupby_name] or data[0][groupby_name] == dt:
    new_data.append(data.popleft())
else:
    new_data.append(dict(empty_data, **{groupby_name: dt}))
```